src/Playwright/Core/BrowserContext.cs

```diff
@@ -326,42 +326,22 @@ public Task<IPage> RunAndWaitForPageAsync(Func<Task> action, BrowserContextRunAn
 
         internal void OnRoute(Route route, IRequest request)
         {
-            foreach (var item in _routes)
+            foreach (var routeHandler in _routes.ToList())
             {
-                if (
-                    ((item.Times ?? 0) == 0 || item.HandledCount >= item.Times) &&
-                    ((item.Url != null && UrlMatches(request.Url, item.Url)) ||
-                    (item.Regex?.IsMatch(request.Url) == true) ||
-                    (item.Function?.Invoke(request.Url) == true)))
+                if (routeHandler.Matches(request.Url))
                 {
-                    item.Handle(route);
+                    if (routeHandler.Handle(route))
+                    {
+                        _routes.Remove(routeHandler);
+                        if (_routes.Count == 0)
+                        {
+                            DisableInterceptionAsync().ConfigureAwait(false);
```

Comment:
```suggestion
                            _ = DisableInterceptionAsync().ConfigureAwait(false);
```

Reply:
Let's make sure to annotate palaces where we need to handle unobserved exception with _ =.